### ISSUE TITLE
Discussion Language breaks `flarum/tags` feature "Hide from All Discussions"

### ISSUE THREAD

**Issue by iPurpl3x** [FriendsOfFlarum/discussion-language#27]
## Bug Report

**Current Behavior**
When using **Discussion Language**, the "Hide from All Discussions" option of `flarum/tags` seems to be ignored.

**Steps to Reproduce**
1. Create a tag with "Hide from All Discussions" enabled
2. Create a discussion in this tag
3. Navigate to "All discussions"
4. See that the created discussion is visible in the list
5. Disable Discussion Language
6. See that now the discussion is hidden, as expected.

**Expected Behavior**
"Hide from All Discussions" should work as expected.

**Environment**
- Flarum version: 1.2.1
- Extension version: 1.2.5
- PHP version: 8

**Comment by iamdarkle** [comment]
I had noticed this problem for a long time and hadn't found the cause 😲 I'm planning to close this other issue (https://github.com/FriendsOfFlarum/discussion-language/issues/17) as it has too many things and is too hard to follow so I will open another one. 

But I continue with the same problems, my apparent conclusion on that issue is the cache, if the "All Discussions" page is cached the filter can be broken, also I could see it very clear when installing PWA recently, the cached version that PWA generates on mobile always shows the broken filter without exceptions. 

You have caught my attention with what you say that the "All Discussions" page is not supposed to have any filter, I would love to go deeper into this in case it would be necessary to improve the extensibility of the core in this sense to end these problems especially because the implementation of the filter in `v17development/flarum-blog` never breaks even with PWA, that makes me rule out that it is for some other extension that plays with the cache (`glowingblue/redis-setup`, `acpl/flarum-lscache`...) and point more clear to the core.
